include/xtensor/xchunked_array.hpp

```diff
@@ -224,10 +230,16 @@ namespace xt
     inline xchunked_array<CS, EX>::xchunked_array(const xexpression<E>& e, S&& chunk_shape)
     {
         resize(e.derived_cast().shape(), std::forward<S>(chunk_shape));
+        assign(e);
+    }
+
+    template <class CS, class EX>
+    template <class E>
+    inline void xchunked_array<CS, EX>::assign(const xexpression<E>& e)
+    {
         xstrided_slice_vector sv(m_chunk_shape.size());  // element slice corresponding to chunk
         std::transform(m_chunk_shape.begin(), m_chunk_shape.end(), sv.begin(),
                        [](auto size) { return range(0, size); });
-
         shape_type ic(this->dimension());  // index of chunk, initialized to 0...
         size_type ci = 0;
         for (auto& chunk: m_chunks)
```

Comment:
I'm not sure if iterating over the chunks is always done in the same order. The code below assumes it is C-order.